pkg/strategy/strategy.go

```diff
@@ -0,0 +1,185 @@
+package strategy
+
+import (
+	osv1 "github.com/openshift/api/route/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	batchv1beta1 "k8s.io/api/batch/v1beta1"
+	"k8s.io/api/core/v1"
+	"k8s.io/api/extensions/v1beta1"
+	rbacv1 "k8s.io/api/rbac/v1"
+)
+
+// S knows what type of deployments to build based on a given spec
+type S struct {
+	typ          Type
```

Comment:
The type is used only in tests. Which could be check based on specific object -eg. all-in-one is present or injester

Reply:
??

Reply:
Sorry, missed this one. This is used in the `jaeger_controller`, inside the `ReconcileJaeger` struct